packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -0,0 +1,324 @@
+		};
+
+		// Create a new object, conditionally including the parent property
+		const updatedArgs = {
+			...args,
+			...( parentCommentId ? { parent: parentCommentId } : {} ),
+		};
+
+		const savedRecord = await saveEntityRecord(
+			'root',
+			'comment',
+			updatedArgs
+		);
+
+		if ( savedRecord ) {
+			// If it's a main comment, update the block attributes with the comment id.
+			if ( ! parentCommentId ) {
+				updateBlockAttributes( clientId, {
+					blockCommentId: savedRecord?.id,
+				} );
+			}
+
+			createNotice(
+				'snackbar',
+				parentCommentId
+					? // translators: Reply added successfully
+					  __( 'Reply added successfully.' )
+					: // translators: Comment added successfully
+					  __( 'Comment added successfully.' ),
+				{
+					type: 'snackbar',
+					isDismissible: true,
+				}
+			);
+			await fetchComments();
+		} else {
+			onError();
+		}
+	};
+
+	const onCommentResolve = async ( commentId ) => {
+		const savedRecord = await saveEntityRecord( 'root', 'comment', {
+			id: commentId,
+			status: 'approved',
+		} );
+
+		if ( savedRecord ) {
+			// translators: Comment resolved successfully
+			createNotice( 'snackbar', __( 'Comment marked as resolved.' ), {
+				type: 'snackbar',
+				isDismissible: true,
+			} );
+
+			await fetchComments();
+		} else {
+			onError();
+		}
+	};
+
+	const onEditComment = async ( commentId, comment ) => {
+		const savedRecord = await saveEntityRecord( 'root', 'comment', {
+			id: commentId,
+			content: comment,
+		} );
+
+		if ( savedRecord ) {
+			createNotice(
+				'snackbar',
+				// translators: Comment edited successfully
+				__( 'Comment edited successfully.' ),
+				{
+					type: 'snackbar',
+					isDismissible: true,
+				}
+			);
+
+			await fetchComments();
+		} else {
+			onError();
+		}
+	};
+
+	const onError = () => {
+		createNotice(
+			'error',
+			// translators: Error message when comment submission fails
+			__(
+				'Something went wrong. Please try publishing the post, or you may have already submitted your comment earlier.'
+			),
+			{
+				isDismissible: true,
+			}
+		);
+	};
+
+	const onCommentDelete = async ( commentId ) => {
+		const childComment = await getEntityRecord(
+			'root',
+			'comment',
+			commentId
+		);
+		await deleteEntityRecord( 'root', 'comment', commentId );
+
+		if ( childComment && ! childComment.parent ) {
+			updateBlockAttributes( clientId, {
+				blockCommentId: undefined,
+			} );
+		}
+
+		createNotice(
+			'snackbar',
+			// translators: Comment deleted successfully
+			__( 'Comment deleted successfully.' ),
+			{
+				type: 'snackbar',
+				isDismissible: true,
+			}
+		);
+		await fetchComments();
+	};
+
+	const fetchComments = useCallback( async () => {
+		if ( postId ) {
+			const data = await getEntityRecords( 'root', 'comment', {
+				post: postId,
+				type: 'block_comment',
+				status: 'any',
+				per_page: 100,
+			} );
+
+			if ( data ) {
+				setThreads( Array.isArray( data ) ? data : [] );
```

Comment:
I would love if this is fixed to avoid the intermediate state.

Reply:
Done

Reply:
Can you point me to the commit that removed the state? As far as I can see, the local `threads` is still here?

Reply:
I would love if this local state and useEffect is avoided, and `threads` is calculated straight from the entity records on render.